devtools/blockreplace.py

```diff
@@ -38,7 +38,7 @@ class Language(str, Enum):
 def replace(filename, blockname, language, content):
     start, stop = comment_style[language]
 
-    tempfile = f"{filename}.tmp"
+    tempfile = "{}.tmp".format(filename)
```

Comment:
nit: python 3.6 is in end of life https://endoflife.date/python I do not see any reasonable way to keep supporting this old version. Python is already enough painful in the supported version

Reply:
@vincenzopalazzo yes. Ubuntu 16.04 (and older not EOL versions) is shipped with Python 3.5 only, and there is no official support for newer versions. This is very popular OS, and the only thing that failing the compilation on this OS is this line. There is no reason not to use `.format` instead of `f-strings`.

Reply:
Ubuntu 16.04 had EOL more than a year ago.

Reply:
> Ubuntu 16.04 had EOL more than a year ago.

@kristapsk, it's not:
![image](https://user-images.githubusercontent.com/4103710/206873687-fef0c952-8a48-4ff0-863e-a0e5790a9097.png)

Reply:
>Ubuntu 16.04 had EOL more than a year ago.

I think there is support for https://ubuntu.com/about/release-cycle but there is no support for python 3.6. 

@sha-265 if it is a popular OS you do not have a problem installing python 3.7.

Reply:
@vincenzopalazzo, there is no official support, and it's not making sense to install it from doubtful sources, on a bitcoin hot wallet machine.

Reply:
at this point, also with python 3.5 you are running vulnerable source code anyway!

Reply:
Yeah, there is no way we're going to lag 10+ years behind on python versions, sorry.